source/agora/node/Ledger.d

```diff
@@ -44,7 +46,15 @@ public class Ledger
     public this (TransactionPool pool)
     {
         this.pool = pool;
+
+        string path = buildPath(getcwd, ".cache");
+        if (!path.exists)
+            mkdir(path);
+        this.ledger = new BlockStorage(path);
+        BlockStorage.removeIndexFile(path);
+
         auto block = getGenesisBlock();
+        this.block_height = size_t.max;
```

Comment:
Why?

Reply:
This initializer will be replaced as soon as the genesis block is created. So it's unnecessary.

Reply:
I can see why this was added. It's because of the code in `isValidBlock` which checks that the header's height is `this.block_height + 1`. However what you are depending on here is integer overflow to wrap around to zero.

It would be much better to just replace the call `this.acceptBlock` with `this.ledger.saveBlock(block);`, and leave `this.block_height` to be initialized with zero (the default).

Reply:
`this.block_height = block.header.height;
        this.storage.saveBlock(block);` 
Applied the suggestion